aws/resource_aws_budgets_budget.go

```diff
@@ -249,8 +422,47 @@ func resourceAwsBudgetsBudgetUpdate(d *schema.ResourceData, meta interface{}) er
 		return fmt.Errorf("update budget failed: %v", err)
 	}
 
+	err = resourceAwsBudgetsBudgetNotificationsUpdate(d, meta)
+
+	if err != nil {
+		return fmt.Errorf("update budget notification failed: %v", err)
+	}
+
 	return resourceAwsBudgetsBudgetRead(d, meta)
 }
+func resourceAwsBudgetsBudgetNotificationsUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+	accountID, budgetName, err := decodeBudgetsBudgetID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	if d.HasChange("notification") {
+		o, n := d.GetChange("notification")
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		removeNotifications, _ := expandBudgetNotificationsUnmarshal(os.Difference(ns).List())
+		addNotifications, addSubscribers := expandBudgetNotificationsUnmarshal(ns.Difference(os).List())
+
+		for _, notification := range removeNotifications {
+			client.DeleteNotification(&budgets.DeleteNotificationInput{
```

Comment:
This API call does not seem to be performing error checking. Also a minor nit, we tend to name our AWS Go SDK clients `conn` across the codebase.

```go
input := &budgets.DeleteNotificationInput{
	AccountId:    aws.String(accountID),
	BudgetName:   aws.String(budgetName),
	Notification: notification,
}

_, err := conn.DeleteNotification(input)

if err != nil {
	return fmt.Errorf("error deleting Budget (%s) Notification: %s", d.Id(), err)
}
```